exonum-java-binding-core/src/main/java/com/exonum/binding/storage/indices/ProofMapIndexProxy.java

```diff
@@ -199,20 +206,64 @@ public V get(K key) {
   private native byte[] nativeGet(long nativeHandle, byte[] key);
 
   /**
-   * Returns a proof that there is a value mapped to the specified key or
-   * that there is no such mapping.
+   * Returns a proof that there are values mapped to the specified keys or that there are no such
+   * mappings.
    *
    * @param key a proof map key which might be mapped to some value, must be 32-byte long
-   * @throws IllegalStateException  if this map is not valid
-   * @throws IllegalArgumentException if the size of the key is not 32 bytes
+   * @param otherKeys other proof map keys which might be mapped to some values, each must be
+   *                  32-byte long
+   * @throws IllegalStateException if this map is not valid
+   * @throws IllegalArgumentException if the size of any of the keys is not 32 bytes
    */
-  public UncheckedMapProof getProof(K key) {
-    byte[] dbKey = keySerializer.toBytes(key);
-    return nativeGetProof(getNativeHandle(), dbKey);
+  public UncheckedMapProof getProof(K key, K... otherKeys) {
+    if (otherKeys.length == 0) {
+      byte[] dbKey = keySerializer.toBytes(key);
+      return nativeGetProof(getNativeHandle(), dbKey);
+    } else {
+      List<K> keys = asList(key, otherKeys);
+      return nativeGetMultiProof(getNativeHandle(), mergeKeysIntoByteArray(keys));
+    }
+  }
+
+  /**
+   * Returns a proof that there are values mapped to the specified keys or that there are no such
+   * mappings.
+   *
+   * @param keys proof map keys which might be mapped to some values, each must be 32-byte long
+   * @throws IllegalStateException if this map is not valid
+   * @throws IllegalArgumentException if the size of any of the keys is not 32 bytes
+   *                                  or keys collection is empty
+   */
+  public UncheckedMapProof getProof(Collection<? extends K> keys) {
+    checkArgument(!keys.isEmpty(), "Keys collection should not be empty");
+    if (keys.size() == 1) {
+      K key = keys.iterator()
+          .next();
+      byte[] dbKey = keySerializer.toBytes(key);
+      return nativeGetProof(getNativeHandle(), dbKey);
+    } else {
+      return nativeGetMultiProof(getNativeHandle(), mergeKeysIntoByteArray(keys));
+    }
   }
 
   private native UncheckedMapProof nativeGetProof(long nativeHandle, byte[] key);
 
+  private byte[] mergeKeysIntoByteArray(Collection<? extends K> keyList) {
```

Comment:
Hm, we could accept Stream here and avoid the intermediate collection, but it won't be enough without the number of keys.